nano/node/node.cpp

```diff
@@ -2727,6 +2727,44 @@ size_t nano::election::last_votes_size ()
 	return last_votes.size ();
 }
 
+void nano::active_transactions::confirm_frontiers (nano::transaction const & transaction_a)
+{
+	// Limit maximum count of elections to start
+	bool representative (node.config.enable_voting && node.wallets.reps_count > 0);
+	/* Check less frequently for non-representative nodes and if previous checks show no unconfirmed frontiers
+	~15 minutes for non-representative nodes, 3.5 minutes for representatives
+	4 times slower if previously all frontiers were confirmed */
+	if (request_confirm_iteration % ((60 - (45 * representative)) * (3 * last_frontier_account.is_zero () + 1)) == 0)
+	{
+		size_t max_elections (max_broadcast_queue / 4);
+		size_t elections_count (0);
+		for (auto i (node.store.latest_begin (transaction_a, last_frontier_account)), n (node.store.latest_end ()); i != n && elections_count < max_elections; ++i)
+		{
+			nano::account_info info (i->second);
+			if (info.block_count != info.confirmation_height)
+			{
+				auto block (node.store.block_get (transaction_a, info.head));
+				if (!start (block))
+				{
+					++elections_count;
+					// Calculate votes for local representatives
+					if (representative)
+					{
+						node.block_processor.generator.add (block->hash ());
+					}
+				}
+				// Update last account
+				last_frontier_account = i->first;
```

Comment:
It looks like this will check the same account the next time through the loop.  last_frontier_account = i->first + 1

Reply:
Right